### PR TITLE
Integrate llvm-project at c50eec400c0edc73eec3c9e97b5c030492cb787f

### PR DESCRIPTION
**Pull request by ThomasRaoux**
* Reset third_party/llvm-project: c50eec400c0edc73eec3c9e97b5c030492cb787f (2022-04-01 10:39:06 -0700): [LLDB] Add require x86 for NativePdb Test.
* reset mhlo to: 48046795afe61e19eef2eb956e06520eb353b77a
* reset TensorFlow to: 5de0f7d48b8d0eb8b430257e8de1c4a7a624e6cd

### CODE REVIEW

File: build_tools/bazel/iree.bazelrc [hunk 1]
```diff
@@ -18,6 +18,9 @@ common --experimental_repo_remote_exec
 # Actually printing output on errors is... a useful default
 test --test_output=errors
 
+build --experimental_cc_shared_library?
```
Comment (ScottTodd) [inline build_tools/bazel/iree.bazelrc:21]: is the question mark intended here?

Also maybe move this up next to https://github.com/google/iree/blob/95eb1e5a9fe7a0e8c67e5b42df4b5b4ebeef5f4f/build_tools/bazel/iree.bazelrc#L16-L17

and add a similar comment. Each setting like this should have some context in a comment so we know roughly why the flag was added. That makes it easier to make changes safely if circumstances change (e.g. if TensorFlow drops the flag or if it is no longer experimental after a certain Bazel version)

File: build_tools/bazel/iree.bazelrc [hunk 2]
```diff
@@ -15,6 +15,7 @@ build --experimental_guard_against_concurrent_changes
 build --compilation_mode=opt
 # Used in TensorFlow, so we have to enable it here as well.
 common --experimental_repo_remote_exec
+build --experimental_cc_shared_library
```
Comment (hanhanW) [inline build_tools/bazel/iree.bazelrc:18]: Looks like we don't need this, because it's added in integrations/tensorflow/build_tools/bazel/iree-tf.bazelrc?

Comment (ThomasRaoux) [reply build_tools/bazel/iree.bazelrc:18]: yes we can remove it. Do you want me to update this PR?

Comment (hanhanW) [reply build_tools/bazel/iree.bazelrc:18]: I'll do it. I was waiting CI bots. It looks fine so far!

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**